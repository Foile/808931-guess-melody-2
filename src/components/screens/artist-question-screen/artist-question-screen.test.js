
import React from 'react';
import renderer from "react-test-renderer";
import ArtistQuestionAcreen from './artist-question-screen';

const questionMock = {
  "type": `artist`,
  "song": {
    "artist": `Quincas Moreira`,
    "src": `https://downloader.disk.yandex.ru/disk/708f2a39b61cd9a01b7e2e6d49de9ffaa831a0dff078d56282d4d1bbc808f5af/5dbb2828/HHdTyAxfdGq5Yrm8EzXG0xdM-9jC3-mWA5TCAiWJUkPDNyuCrmoIZArc2TYKYtAyHH3NeQQjZBMKbim3SfUI6g%3D%3D?uid=0&filename=Three%20Days%20Grace%20-%20Animal%20I%20Have%20Become.mp3&disposition=attachment&hash=xQM7/CqqnBR9OF/Ar/mH91rH8K8v/mxWWsUgMW2U9LQj2urSoX7HiIRCyh3cFSOzq/J6bpmRyOJonT3VoXnDag%3D%3D&limit=0&content_type=audio%2Fmpeg&owner_uid=25986678&fsize=9343104&hid=5d4ec3ef32a9b2630887c70fc9bb8449&media_type=audio&tknv=v2`
  },
  "answers": [
    {
      "id": `1`,
      "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
      "artist": `Quincas Moreira`
    },
    {
      "id": `2`,
      "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
      "artist": `Density & Time`
    },
    {
      "id": `3`,
      "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
      "artist": `Endless Love`
    }
  ]
};


it(`ArtistQuestionAcreen correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<ArtistQuestionAcreen
    question={questionMock}
    screenIndex={0}
    onAnswer={jest.fn()}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
