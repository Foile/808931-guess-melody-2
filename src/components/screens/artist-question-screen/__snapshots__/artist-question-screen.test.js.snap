// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionAcreen correctly renders after relaunch 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          type="radio"
          value="answer-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="Quincas Moreira"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
          />
          Quincas Moreira
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="answer-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Density & Time"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
          />
          Density & Time
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value="answer-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Endless Love"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
          />
          Endless Love
        </label>
      </div>
    </form>
  </section>
</section>
`;
